src/engines.jl

```diff
@@ -61,14 +64,56 @@ get_test_engine()::String = TEST_ENGINE_PROVISION.provider()
 # Release test engine. Notifies the provider that this engine is no longer in use.
 release_test_engine(name::String) = TEST_ENGINE_PROVISION.releaser(name)
 
+"""
+    set_engine_name_provider(provider::Function)
+
+Set a provider for test engine names.
+
+The provider is called by each test to select an engine to run the test with. The default
+provider selects a name from a pool of available test engines.
+
+# Examples
+
+```
+set_engine_name_provider(() -> "MyEngine")
+set_engine_name_provider(() -> create_default_engine("MyEngine"))
```

Comment:
do we really want to encourage usage of passing a create engine function to the engine name provider? It feels a bit off (though I may well be missing somthing)

Reply:
Good point, that could easily go wrong. The example works, since it will create the engine once and once only, but that's specific to this implementation. I'll change it to
```suggestion
set_engine_name_provider(() -> my_custom_engine_selector("MyEngine"))
```